pkg/streamingpromql/operators/aggregations/common.go

```diff
@@ -35,3 +37,27 @@ var AggregationGroupFactories = map[parser.ItemType]AggregationGroupFactory{
 //
 // Invalid combinations include exponential and custom buckets, and histograms with incompatible custom buckets.
 var invalidCombinationOfHistograms = &histogram.FloatHistogram{}
+
+func GetAggregationItemType(aggregation string) (parser.ItemType, bool) {
+	// The parser key is not exported, but the item types are. It's safe to assume the keys will continue to map their names onto their values.
+	// (ie, "avg" will be parser.AVG).
+	var key = map[string]parser.ItemType{
```

Comment:
There's no need to create this map each time `GetAggregationItemType` is called - what if we move this outside this method?